superset-frontend/src/explore/components/controls/DateFilterControl/components/CurrentCalendarFrame.tsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect } from 'react';
+import { t } from '@superset-ui/core';
+import { Radio } from 'src/components/Radio';
+import {
+  CURRENT_CALENDAR_RANGE_OPTIONS,
+  CURRENT_CALENDAR_RANGE_SET,
+} from 'src/explore/components/controls/DateFilterControl/utils';
+import {
+  CurrentCalendarRangeType,
+  CurrentCalendarWeek,
+  FrameComponentProps,
+} from '../types';
+
+export function CurrentCalendarFrame({ onChange, value }: FrameComponentProps) {
+  useEffect(() => {
```

Comment:
**Issue**: The useEffect hook does not correctly handle the dependency array, potentially leading to incorrect or excessive updates. <br> **Fix**: Ensure that the dependency array of the useEffect hook correctly lists all dependencies used within the hook to avoid unnecessary re-renders or stale data issues. <br> **Code Suggestion**: 
 ```
 -  }, [onChange, value]);
 +  }, [value]);
 ```